### ISSUE TITLE
Support for PEP508

### ISSUE THREAD

**Issue by escaped** [sdispater/poetry#21]
With PEP508 you can limit certain package to a specific platforms, e.g. `waitress==1.0.2;platform_system=="Windows"` to install `waitress` only on Windows.

It seem like `poetry` doesn't support this feature at all or at least i wasn't able to specify the constraint from above. Neither `poetry add waitress==1.0.2;platform_system=="Windows"` nor writing `waitress = "=1.0.2;platform_system=Windows"` to the `pyproject.toml` works.

**Comment by escaped** [comment]
Hey @sdispater,

i wonder why you closed this issue. I checked the recent release and could not find anything related to this issue except for `extras`, which do not include plattform specific identifiers. I still think it would be really nice to automatically decide whether a required dependency should be installed. Examples for such packages would be:
* `pyinotify`, which works only on linux
* `enum34`, which is required for python <3.4

**Comment by jacebrowning** [comment]
I would use this for `pync`, which only works on macOS.

**Comment by sdispater** [comment]
The `sys_platform` constraint is supported by writing your dependency like this:

```toml
my-dependency = { version: "*",  "platform": "darwin"}
```

Note, however, that due to a bug it will be lost if/when packaging/publishing your project. This will likely be fixed in the `0.10.0` release.

**Comment by jacebrowning** [comment]
With `0.10.0-alpha.1`, this line in `pyproject.toml`:

```toml
pync = { version = '*', platform = 'darwin' }
```

causes this error with `$ poetry lock`:

```sh
[InvalidProjectFile]
[dev-dependencies.pync] {'version': '*', 'platform': 'darwin'} is not valid under any of the given schemas
```

**Comment by sdispater** [comment]
@jacebrowning It should be fixed in the latest `0.10.0a2` prerelease.
